metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala

```diff
@@ -434,12 +437,18 @@ object MetalsEnrichments
       filename.endsWith(".jar") || filename.endsWith(".srcjar")
     }
 
+    def isSourceJar: Boolean =
+      path.jarPath.exists(_.filename.endsWith("-sources.jar"))
+
+    def isJDKSource: Boolean =
+      path.jarPath.exists(_.filename == "src.zip")
```

Comment:
There functions are correct in 99% percents but I believe it might cause some exceptionally rare issues that would be hard to investigate.

There is non zero possibility that there might a be a dependency jar without a version that ends with `-sources` in build - some local unmanaged jar.

The same for `src.zip` - probably there might be some `src.zip` that aren't about Jdk at all. Also, I'm not sure if all jdk/os distros put sources in `src.zip`

It would be more reliable always check it if jar is a source in `BuildTargets` as it's anyway quite fast.